### PR TITLE
test: add assign, update and complete ITs for user task audit log

### LINKED ISSUE #41238: Add user task audit log acceptance tests
### 

### Description

Add acceptance test:
- User task audit log entries can be queried with the `POST /user-tasks/{userTaskKey}/audit-logs/search` endpoint
- The following operations should have test cases:
   - User task assignment
   - User task update
   - User task complete

### Motivation

Ensure we have E2E acceptance tests of the user task audit log implementation.

### Acceptance criteria

- [ ] ...

### Links

_No response_

### Breakdown

_No response_

### PR DESCRIPTION
**Pull request by tanjinyi**
## Description

Extends the audit log user task search ITs to check for assignment, update and delete user task audit logs separately.

## Checklist

- [ ] Enable backports when necessary (fex. [for bug fixes](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#backporting-changes) or [for CI changes](https://camunda.github.io/camunda/ci/#when-to-backport-ci-changes)).

## Related issues

closes #41238

### CODE REVIEW

File: qa/acceptance-tests/src/test/java/io/camunda/it/auditlog/AuditLogSearchClientIT.java [hunk 1]
```diff
@@ -164,4 +169,52 @@ void shouldSearchUserTaskAuditLogByKeyWithActorId(
     assertThat(auditLog.getCategory()).isEqualTo(AuditLogCategoryEnum.USER_TASKS);
     assertThat(auditLog.getResult()).isEqualTo(AuditLogResultEnum.SUCCESS);
   }
+
+  @Test
+  void shouldSearchUserTaskAuditLogByKeyWithUpdateOperation(
```
Comment (eppdot) [inline qa/acceptance-tests/src/test/java/io/camunda/it/auditlog/AuditLogSearchClientIT.java]: ❌ let's put the tracking focused ITs in it's own file, e.g. see `AuditLogIdentityOperationsIT`. This IT is intended to check the search with focus on filter combinations and edge cases.

File: qa/acceptance-tests/src/test/java/io/camunda/it/auditlog/AuditLogSearchClientIT.java [hunk 2]
```diff
@@ -142,7 +147,7 @@ void shouldSearchUserTaskAuditLogByKey(
   }
 
   @Test
-  void shouldSearchUserTaskAuditLogByKeyWithActorId(
+  void shouldSearchUserTaskAuditLogByKeyWithAssignOperation(
```
Comment (eppdot) [inline qa/acceptance-tests/src/test/java/io/camunda/it/auditlog/AuditLogSearchClientIT.java]: 🔧 searching by actor should remain here, as this is a filtering focussed test and not operation tracking. see other comment :)

File: qa/acceptance-tests/src/test/java/io/camunda/it/auditlog/AuditLogUtils.java [hunk 3]
```diff
@@ -149,6 +156,61 @@ public void assignUserToUserTask(final String username, final Long processInstan
...
+                      .send()
+                      .join();
+              assertThat(auditLogItems.items()).hasSize(1);
+            });
+  }
+
+  public void completeUserTask(final UserTaskRecordValue userTask) {
+    defaultClient.newCompleteUserTaskCommand(userTask.getUserTaskKey()).send().join();
+
+    Awaitility.await("Audit log entry is created for the assigned user task")
```
[the remark this answers is not in the document]
Comment (eppdot) [reply qa/acceptance-tests/src/test/java/io/camunda/it/auditlog/AuditLogUtils.java]: sounds valid 😄

File: qa/acceptance-tests/src/test/java/io/camunda/it/auditlog/AuditLogUtils.java [hunk 4]
```diff
@@ -160,7 +222,11 @@ public AuditLogUtils generateDefaults() {
 
     deployResource("B.bpmn", PROCESS_B, TENANT_A);
     final var instanceWithUserTask = startProcessInstance(PROCESS_B_ID, TENANT_A);
-    assignUserToUserTask(DEFAULT_USERNAME, instanceWithUserTask.getProcessInstanceKey());
+    final UserTaskRecordValue completedUserTaskB =
```
Comment (eppdot) [inline qa/acceptance-tests/src/test/java/io/camunda/it/auditlog/AuditLogUtils.java]: ❌ this should only be done in your test setup, but not for all audit log tests - which would be the case adding it to this utils setup. Thus, the request to put the user task ITs in its own file will help also to have a clear `setUp` behavior.

**Review by eppdot — CHANGES_REQUESTED** [review]
👍 overall looks solid with clear test scenarios.

❌ But let's move the operation tracking tests into it's own file

**Review by eppdot — APPROVED** [review]
👍 looks good to me. Should we also test the rejection intent etc?

**Comment by tanjinyi** [comment]
> 👍 looks good to me. Should we also test the rejection intent etc?

hmm, it would probably be good to have. maybe we can address this later on
